src/libstd/sys_common/process.rs

```diff
@@ -121,4 +125,12 @@ impl<K: EnvKey> CommandEnv<K> {
         self.clear = true;
         self.vars.clear();
     }
+    pub fn have_changed_path(&self) -> bool {
+        self.saw_path || self.clear
+    }
+    fn maybe_saw_path(&mut self, key: &OsStr) {
+        if !self.saw_path && key.to_os_string() == OsString::from("PATH") {
```

Comment:
I think here you should be able to do a direct comparison with `key == "PATH"`?

Reply:
Yeah you're right. Weird, I fought with this for far too long :).

Reply:
Fix pushed.